### PR TITLE
support debug each output of each ins

### PR DESCRIPTION
**Pull request by Thunderbrook**
In debug mode, we need to know each output, so in this commit support to dump each output tensor of each instance into afs.

### CODE REVIEW

File: python/paddle/fluid/incubate/fleet/parameter_server/pslib/__init__.py [hunk 1]
```diff
@@ -485,7 +501,7 @@ def minimize(self,
                           parameter_list,
                           no_grad_set,
                           self._strategy)
-
+        opt_info["mpi_rank"] = fleet._role_maker._get_rank()
```
Comment (xjqbest) [inline python/paddle/fluid/incubate/fleet/parameter_server/pslib/__init__.py:577]: maybe trainer_rank or trainer_id is better，role_maker can have many types (mpi or paddlecloud, etc.)  and  all of them can get_rank

File: paddle/fluid/framework/downpour_worker.cc [hunk 2]
```diff
@@ -72,6 +74,85 @@ void DownpourWorker::Initialize(const TrainerDesc& desc) {
...
+  }
+  need_dump_field_ = false;
+  if (dump_fields_.size() != 0 && dump_fields_path_ != "") {
+    need_dump_field_ = true;
+  }
+  mpi_rank_ = desc.mpi_rank() / 2;
+}
+
+template <typename T>
+std::string print_lod_tensor_type(LoDTensor* tensor, int64_t start,
```
Comment (xjqbest) [inline paddle/fluid/framework/downpour_worker.cc]: code style of function name is wrong,  so as other functions

Comment (Thunderbrook) [reply paddle/fluid/framework/downpour_worker.cc]: done

**Review by yaoxuefeng6 — APPROVED** [review]
LGTM

**Review by xjqbest — APPROVED** [review]

**Pull request merged**